hyperspy/axes.py

```diff
@@ -524,10 +524,23 @@ def _parse_value_from_string(self, value):
         """Return calibrated value from a suitable string """
         if len(value) == 0:
             raise ValueError("Cannot index with an empty string")
+        # Starting with 'rel', it must be relative slicing
+        elif value.startswith('rel'):
+            try:
+                relative_value = float(value[3:])
+            except ValueError:
+                raise ValueError("`rel` must be followed by a number in range [0, 1].")
+            if relative_value < 0 or relative_value > 1:
+                raise ValueError("Relative value must be in range [0, 1]")
+            value = self.low_value + relative_value * (self.high_value - self.low_value)
         # if first character is a digit, try unit conversion
         # otherwise we don't support it
         elif value[0].isdigit():
-            value = self._get_value_from_value_with_units(value)
+            if self.is_uniform:
+                value = self._get_value_from_value_with_units(value)
+            else:
+                raise ValueError("Unit conversion is only supported for "
+                                 "uniform axis.")
```

Comment:
To me this looks good.

We just need add `value = self._parse_value(value)` at the beginning of `FunctionalDataAxis.value2index(...)`

Should I include it in #2743 ? I can also add this commit in #2743 directly, maybe it's the easiest no?

Reply:
I would keep these PR separated, to keep them as simple as possible, which is not only useful now but also in the future, when looking back at it!

Reply:
:+1:

Reply:
I do not see the proposed line added in either of the two PRs. Is this resolved anyway?

Reply:
I suggested to do bit in another PR: remove `UniformDataAxis.value2index` and use `BaseDataAxis.value2index` instead as discussion in https://github.com/hyperspy/hyperspy/pull/2743#discussion_r634244400.